storage/remote/queue_manager.go

```diff
@@ -551,11 +631,72 @@ outer:
 	return true
 }
 
-func (t *QueueManager) AppendExemplars(exemplars []record.RefExemplar) bool {
+func (t *QueueManager) updatePendingData(dataCount, dataSegment int) {
+	t.pendingDataMut.Lock()
+	defer t.pendingDataMut.Unlock()
+
+	t.pendingDataSegments[dataSegment] += dataCount
+
+	// We want to save segments whose data has been fully processed by the
+	// QueueManager. To avoid too much overhead when appending data, we'll only
+	// examine pending data per segment whenever a new segment is detected.
+	//
+	// This could cause our saved segment to lag behind multiple segments
+	// depending on the rate that new segments are created and how long
+	// data in other segments takes to be processed.
+	if dataSegment < t.lastMarkedSegment {
```

Comment:
in what case would this condition ever be true?

Reply:
Oops, this should actually be `dataSegment <= t.latestDataSegment`